### PR TITLE
VideoCore: Unify interface to OpenGL and SW rasterizers

### PR DESCRIPTION
**Pull request by yuriks**
This removes explicit checks sprinkled all over the codebase to instead just have the SW rasterizer expose an implementation with no-ops for most operations.

Also does a few other cleanups in that interface.

### CODE REVIEW

**Comment by yuriks** [comment]
Updated.

File: src/video_core/renderer_opengl/gl_rasterizer.h [hunk 1]
```diff
@@ -123,16 +123,16 @@ class RasterizerOpenGL : public HWRasterizer {
     void DrawTriangles() override;
 
     /// Commit the rasterizer's framebuffer contents immediately to the current 3DS memory framebuffer
-    void CommitFramebuffer() override;
+    void FlushFramebuffer() override;
 
     /// Notify rasterizer that the specified PICA register has been changed
     void NotifyPicaRegisterChanged(u32 id) override;
 
     /// Notify rasterizer that the specified 3DS memory region will be read from after this notification
```
Comment (tfarley) [inline src/video_core/renderer_opengl/gl_rasterizer.h]: Comment should be updated to match rasterizer_interface.h (also comment for InvalidateRegion below)

**Comment by yuriks** [comment]
D-d-double update!

**Comment by Subv** [comment]
LGTM

**Comment by linkmauve** [comment]
LGTM.

**Comment by bunnei** [comment]
LGTM

**Pull request merged**